projects/wave-core-new/src/lib/operators/dialogs/raster-vector-join/raster-vector-join.component.ts

```diff
@@ -0,0 +1,219 @@
+                    if (!this.valueNameUserChanges[i]) {
+                        (valueNames.at(i) as FormControl).setValue(rasters[i].name, {emitEvent: false});
+                    }
+                }
+            }
+
+            this.reCheckValueNames();
+        });
+        this.subscriptions.push(rasterLayersSubscription);
+    }
+
+    /**
+     * Checks for collisions of value name.
+     * Uses `startsWith` semantics.
+     */
+    private valueNameCollision(valueNames: FormArray) {
+        return (control: FormControl): { [key: string]: boolean } => {
+            const errors: {
+                duplicateName?: boolean,
+            } = {};
+
+            const valueName: string = control.value;
+
+            let duplicates = -1; // subtracting self
+
+            for (const otherValueName of valueNames.value as Array<string>) {
+                if (otherValueName === valueName) {
+                    duplicates++;
+                }
+            }
+
+            if (duplicates < 1) { // check for conflicts with extisting column in input
+                for (const otherValueName of this.vectorColumns) {
+                    if (otherValueName === valueName) {
+                        duplicates++;
+                        break;
+                    }
+                }
+            }
+
+            if (duplicates > 0) {
+                errors.duplicateName = true;
+            }
+
+            return Object.keys(errors).length > 0 ? errors : null;
+        };
+    }
+
+    add() {
+        const vectorLayer: VectorLayer = this.form.controls['vectorLayer'].value;
+        const rasterLayers: Array<RasterLayer> = this.form.controls['rasterLayers'].value;
+        const valueNames: Array<string> = this.form.controls['valueNames'].value;
+        const aggregation: Aggregation = this.form.controls['aggregation'].value;
+
+        const outputLayerName: string = this.form.controls['name'].value;
+
+        const params: RasterVectorJoinParams = {
+            names: valueNames,
+            aggregation,
+        };
+
+        const workflowIds = [
+            vectorLayer.workflowId,
+            ...rasterLayers.map(rasterLayer => rasterLayer.workflowId),
+        ];
+        combineLatest(workflowIds.map(workflowId => this.projectService.getWorkflow(workflowId))).pipe(
+            mergeMap(([vectorWorkflow, ...rasterWorkflows]) => this.projectService.registerWorkflow({
+                type: 'Vector',
+                operator: {
+                    type: 'RasterVectorJoin',
+                    params,
+                    vector_sources: [vectorWorkflow.operator],
+                    raster_sources: rasterWorkflows.map(workflow => workflow.operator),
+                }
+            })),
+            mergeMap(workflowId => this.projectService.addLayer(new VectorLayer({
+                workflowId,
+                name: outputLayerName,
+                symbology: vectorLayer.symbology,
```

Comment:
do we really want to use the same symbology? In our demos we always showed that the new points in a new color now have the raster values attached

Reply:
c9c2f55